client/my-sites/checkout/utils/use-equivalent-monthly-totals.ts

```diff
@@ -0,0 +1,59 @@
+import {
+	isMonthly,
+	isWpComPlan,
+	getMonthlyPlanByYearly,
+	type PlanSlug,
+} from '@automattic/calypso-products';
+import { Plans } from '@automattic/data-stores';
+import { ResponseCartProduct } from '@automattic/shopping-cart';
+import { useMemo } from 'react';
+import useCheckPlanAvailabilityForPurchase from 'calypso/my-sites/plans-features-main/hooks/use-check-plan-availability-for-purchase';
+
+/**
+ * Calculate the equivalent monthly total prices for non-monthly WordPress.com plans from the passed
+ * cart. For each eligible product it'd determine the equivalent monthly plan and return the monthly
+ * plan price multiplied by the number of months in the yearly (or longer) plan.
+ * Used for demonstrating the benefits of purchasing longer-term plans.
+ *
+ * @param products - Array of `ResponseCartProduct` items.
+ *
+ * @returns An object where each key is an eligible plan slug from the cart and the value is the
+ *          calculated total cost as if it was billed monthly.
+ */
+export default function useEquivalentMonthlyTotals(
+	products: Array< ResponseCartProduct >
```

Comment:
Nit - In general, I find it better to create single-parameter functions (an object param), so it's easier to extend or down the line. It's a common pattern in React e.g. the resulting type being

```
interface UseEquivalentMonthlyTotalsProps {
	products: Array< ResponseCartProduct >;
}

type UseEquivalentMonthlyTotals = (
	props: UseEquivalentMonthlyTotalsProps
) => Record< PlanSlug, number >;

```